example/glide_computer.cpp

```diff
@@ -24,151 +24,45 @@
-    requires requires(Q q) { q -= v.magnitude(); }
-  {
-    magnitude_ -= v.magnitude();
-    return *this;
-  }
-
-  template<typename Q2>
-  [[nodiscard]] friend constexpr auto operator+(const vector& lhs, const vector<Q2, D>& rhs)
-    requires requires { lhs.magnitude() + rhs.magnitude(); }
-  {
-    using ret_type = decltype(lhs.magnitude() + rhs.magnitude());
-    return vector<ret_type, D>(lhs.magnitude() + rhs.magnitude());
-  }
-
-  template<typename Q2>
-  [[nodiscard]] friend constexpr auto operator-(const vector& lhs, const vector<Q2, D>& rhs)
-    requires requires { lhs.magnitude() - rhs.magnitude(); }
-  {
-    using ret_type = decltype(lhs.magnitude() - rhs.magnitude());
-    return vector<ret_type, D>(lhs.magnitude() - rhs.magnitude());
-  }
-
-  template<typename V>
-    requires (QuantityValue<V> || Dimensionless<V>)
-  [[nodiscard]] friend constexpr auto operator*(const vector& lhs, const V& value)
-    requires requires { lhs.magnitude() * value; }
-  {
-    return vector<Q, D>(lhs.magnitude() * value);
-  }
-
-  template<typename V>
-    requires (QuantityValue<V> || Dimensionless<V>)
-  [[nodiscard]] friend constexpr auto operator*(const V& value, const vector& rhs)
-    requires requires { value * rhs.magnitude(); }
-  {
-    return vector<Q, D>(value * rhs.magnitude());
-  }
-
-  template<typename Q2, direction D2>
-  [[nodiscard]] friend constexpr auto operator/(const vector& lhs, const vector<Q2, D2>& rhs)
-    requires requires { lhs.magnitude() / rhs.magnitude(); }
-  {
-    return lhs.magnitude() / rhs.magnitude();
-  }
-
-  template<typename Q2>
-  [[nodiscard]] friend constexpr auto operator<=>(const vector& lhs, const vector<Q2, D>& rhs)
-    requires requires { lhs.magnitude() <=> rhs.magnitude(); }
-  {
-    return lhs.magnitude() <=> rhs.magnitude();
-  }
-
-  template<typename Q2>
-  [[nodiscard]] friend constexpr bool operator==(const vector& lhs, const vector<Q2, D>& rhs)
-    requires requires { lhs.magnitude() == rhs.magnitude(); }
-  {
-    return lhs.magnitude() == rhs.magnitude();
-  }
-
-  template<class CharT, class Traits>
-    requires Quantity<Q>
-  friend std::basic_ostream<CharT, Traits>& operator<<(std::basic_ostream<CharT, Traits>& os, const vector& v)
-  {
-    return os << v.magnitude();
-  }
-
-private:
-  Q magnitude_{};
-};
-
-
-template<typename T>
-inline constexpr bool is_vector = false;
-template<typename Q, direction D>
-inline constexpr bool is_vector<vector<Q, D>> = true;
-
-} // namespace
-
-template<Quantity Q, direction D>
-struct fmt::formatter<vector<Q, D>> : formatter<Q> {
+template<units::QuantityKind QK>
+struct fmt::formatter<QK> : formatter<typename QK::quantity_type> {
   template <typename FormatContext>
-  auto format(const vector<Q, D>& v, FormatContext& ctx)
+  auto format(const QK& v, FormatContext& ctx)
   {
-    return formatter<Q>::format(v.magnitude(), ctx);
+    return formatter<typename QK::quantity_type>::format(v.common(), ctx);
```

Comment:
By the way, I got this when using `fmt`'s `master` branch. The `quantity` formatter uses a removed `fmt::detail::` name.
```
../../../example/glide_computer.cpp:144:17:   required from ‘void {anonymous}::print(const R&) [with R = std::array<{anonymous}::glider, 4>]’
../../../example/glide_computer.cpp:380:16:   required from here
/home/johel/Documents/C++/Forks/mpusz/units/src/src/include/units/format.h:91:73: error: ‘make’ is not a member of ‘fmt::v7::detail::fill_t<char>’
```

Reply:
Actually, that was using my system-wide `fmt`, the latest stable release at the time.